packages/protect/src/schema/index.ts

```diff
@@ -0,0 +1,243 @@
+import { z } from 'zod'
+
+// ------------------------
+// Zod schemas
+// ------------------------
+const castAsEnum = z
+  .enum([
+    'big_int',
+    'boolean',
+    'date',
+    'real',
+    'double',
+    'int',
+    'small_int',
+    'text',
+    'jsonb',
+  ])
+  .default('text')
+
+const tokenFilterSchema = z.object({
+  kind: z.literal('downcase'),
+})
+
+const tokenizerSchema = z
+  .union([
+    z.object({
+      kind: z.literal('standard'),
+    }),
+    z.object({
+      kind: z.literal('ngram'),
+      token_length: z.number(),
+    }),
+  ])
+  .default({ kind: 'standard' })
```

Comment:
I think that `ngram` might be a more useful default. Might want to use the same as Proxy (ngram with token length of 3). Proxy also defaults to using a downcase token filter, too (which basically makes search case insensitive).

Reply:
Megan is actually the default on the higher order interface. I'll clean up the Zod schema

Reply:
Agreed. `ngram` is more versatile for most scenarios.